src/utils.c

```diff
@@ -1354,6 +1395,9 @@ void vctrs_init_utils(SEXP ns) {
 
   base_method_table = r_env_get(R_BaseNamespace, Rf_install(".__S3MethodsTable__."));
 
+  s4_c_method_table = r_parse_eval("environment(methods::getGeneric('c'))$.MTable", R_GlobalEnv);
+  R_PreserveObject(s4_c_method_table);
```

Comment:
Is the right thing to do here to preserve this object?

Reply:
It shouldn't hurt. If the table is replaced for some reason and gets GC'd, we'd be working with outdated methods, but we wouldn't crash. Since I don't know anything about S4 I'm not sure if that's a possibility, but I'd be surprised if it were.